### PR TITLE
[FSDE-30732] Fix See More on load tabs

### PR DESCRIPTION
**Pull request by lumdmtr**
https://ctl.atlassian.net/browse/FSDE-30732

### CODE REVIEW

File: src/custom-elements/src/components/tabs/tabs.tsx [hunk 1]
```diff
@@ -363,6 +363,25 @@ export class Tabs {
     }
   }
 
+  setAvailableSpaceAsync() {
+    return new Promise((resolve) => {
+      const startTime = Date.now();
+      const waitLimit = 1000; // milliseconds
```
Comment (krzysztofdrozdowskilumen) [inline src/custom-elements/src/components/tabs/tabs.tsx:369]: I think 'timeout' would be more descriptive.

File: src/custom-elements/src/components/tabs/tabs.tsx [hunk 2]
```diff
@@ -363,6 +363,25 @@ export class Tabs {
...
+        const { height, width } = this.el.getBoundingClientRect();
+        // fallback for case when height or width equals 0
+        const isWaitingTooLong = (Date.now() - startTime)  > waitLimit;
+        if (height && width || isWaitingTooLong) {
+          this.setAvailableSpace();
+          resolve();
+          return;
+        }
+        asyncCheck();
+      }, 0);
```
Comment (krzysztofdrozdowskilumen) [inline src/custom-elements/src/components/tabs/tabs.tsx:380]: The second argument of setTimeout is optional. You can skip it.

File: src/custom-elements/src/components/tabs/tabs.tsx [hunk 3]
```diff
@@ -363,6 +363,25 @@ export class Tabs {
...
   }
 
+  setAvailableSpaceAsync() {
+    return new Promise((resolve) => {
+      const startTime = Date.now();
+      const waitLimit = 1000; // milliseconds
+      const asyncCheck = () => setTimeout(() => {
+        const { height, width } = this.el.getBoundingClientRect();
+        // fallback for case when height or width equals 0
+        const isWaitingTooLong = (Date.now() - startTime)  > waitLimit;
```
Comment (krzysztofdrozdowskilumen) [inline src/custom-elements/src/components/tabs/tabs.tsx:373]:
```suggestion
        const isWaitingTooLong = (Date.now() - startTime) > waitLimit;
```
Unnecessary space.

**Review by krzysztofdrozdowskilumen — CHANGES_REQUESTED** [review]
Great job 👏 I have only few comments:

File: src/custom-elements/src/components/tabs/tabs.tsx (hunk 1 above)
Comment (lumdmtr) [reply src/custom-elements/src/components/tabs/tabs.tsx:369]: Thanks for pointing it out! @krzysztofdrozdowskilumen 

Actually, it's not a 'timeout'.
It's time limit, so the function is executed no more than the time limit.

To my mind, 'timeout' is more about an opposite - delay, break or suspension.

Probably, `limit`, `timeLimit`, `executionLimit` or `maxWorkTime` would be more descriptive
But :laughing:

**Review by krzysztofdrozdowskilumen — APPROVED** [review]
🙌

**Pull request closed without merging**